### ISSUE TITLE
Better NVM integration

### ISSUE THREAD

**Issue by szszoke** [microsoft/vscode#142607]
**_TL;DR;_**
```
As a user, I want to set `$HOME/.nvm` as the NVM directory in my VS Code settings
And have VS Code do the necessary steps to load NVM
So that VS Code is aware of my NVM configuration even if it is started from a GUI shortcut
```

**_Long version_**
NVM is very useful when one has to work with different Node.js versions. It enables setting a specific Node.js version for a console session. In order for this to work, a few commands need to be run to load NVM. This is typically achieved by putting the needed commands into `.bashrc`/`.zshrc`/etc. which are then executed by the shell when a new console session is started.

```
export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
```
Applications that are started from the resulting console session would then be aware of Node.js version configured via NVM.

There is a catch however: one must always run those said applications from a console session where NVM was loaded.

In VS Code's case, when the it is launched via a GUI shortcut, NVM is never loaded and VS Code stays ignorant about it.

This will cause issues like #131911 where the configured Node.js version is seemingly ignored. Another extension that is sensitive to the Node.js version is for Azure Functions.

There are at least two scenarios that I know of where VS Code is not launched from a console with NVM loaded:
- user launches VS Code from a GUI shortcut
- VS Code is launched from the browser (opening a Live Share session for example)

It would be nice if I could point VS Code to my NVM directory and have it load my configuration when it starts up.

There is already some support for specifying the Node.js version for `node` tasks but this would make it better.

**Comment by Tyriar** [comment]
> In VS Code's case, when the it is launched via a GUI shortcut, NVM is never loaded and VS Code stays ignorant about it.

VS Code does source a _login_ shell when you launch it from a shortcut, you could try add the nvm config to your profile script instead and/or source your .bashrc file from the profile script.

**Comment by szszoke** [comment]
My system is configured to use ZSH. I have the lines that load NVM in my .zshrc file, and if start a terminal application then NVM is loaded.

VS Code is also configured to use ZSH.

If I start VS Code from that terminal then I don't have issues with VS Code using a different Node.js version then what is set to default by NVM.

Would VS Code know that it has to source my .zshrc file and not something else?

**Comment by Tyriar** [comment]
@szszoke vscode sources a login shell, adding `. ~/.zshrc` to your `~/.zprofile` (or `~/.profile`?) should get it to show up.

**Comment by oscarmarcelo** [comment]
I have the same issue!

I use zsh and I have everything configured, including [automatically](https://github.com/nvm-sh/nvm#calling-nvm-use-automatically-in-a-directory-with-a-nvmrc-file) setting up the correct Node version if `.nvmrc`.

If I open a new terminal, the profile loads correctly, but if I try to run a NPM script as a task (clicking in the *Run* button in the NPM Scripts panel), the script will run right away without setting up the profile first, thus breaking the script that depends on the Node version defined in `.nvmrc`.

Funny thing is that if I instead *Debug* the same task (clicking in the button next to *Run*), the profile will now run before running the actual task...

**Comment by szszoke** [comment]
> @szszoke vscode sources a login shell, adding `. ~/.zshrc` to your `~/.zprofile` (or `~/.profile`?) should get it to show up.

Indeed you were right.

I first tried to put the NVM loader into `~/.profile` and then reboot. That didn't work.

I then symlinked `~/.profile` to `~/.zprofile` and now VS Code uses the expected Node.js version when starting tasks.

Thank you!
